posydon/unit_tests/utils/test_data_download.py

```diff
@@ -0,0 +1,235 @@
+            assert failed_MD5_statement in captured_output.out
+            assert extraction_statement in captured_output.out
+            assert removal_statement not in captured_output.out
+            # without MD5 check
+            totest.data_download(file=test_path+".tar.gz", MD5_check=False)
+            captured_output = capsys.readouterr()
+            assert download_statement in captured_output.out
+            assert failed_MD5_statement not in captured_output.out
+            assert extraction_statement in captured_output.out
+            assert removal_statement not in captured_output.out
+        # skip real download: create mock file instead
+        with monkeypatch.context() as mp:
+            mp.setattr(totest.urllib.request, "urlretrieve", mock_urlretrieve2)
+            # mocked download: fails MD5check, which removes tar file and
+            #                  causes a FileNotFoundError
+            with raises(FileNotFoundError, match="No such file or directory:"):
+                totest.data_download(file=test_path+"0.tar.gz", verbose=True)
+            captured_output = capsys.readouterr()
+            assert download_statement in captured_output.out
+            assert failed_MD5_statement in captured_output.out
+            assert extraction_statement in captured_output.out
+            assert removal_statement not in captured_output.out
+            # return expected hash for testfile
+            with patch("hashlib.md5") as p_md5:
+                p_md5.return_value.hexdigest.return_value = totest.original_md5
+                totest.data_download(file=test_path+"1.tar.gz")
+                captured_output = capsys.readouterr()
+                assert download_statement in captured_output.out
+                assert failed_MD5_statement not in captured_output.out
+                assert extraction_statement in captured_output.out
+                assert removal_statement not in captured_output.out
+                assert totest.os.path.exists(test_path)
+                assert totest.os.path.exists(totest.os.path.join(test_path,\
+                                                                 "test1.txt"))
+                rmtree(test_path) # removed extracted data
+                # with verification output
+                totest.data_download(file=test_path+"2.tar.gz", verbose=True)
+                captured_output = capsys.readouterr()
+                assert download_statement in captured_output.out
+                assert "MD5 verified" in captured_output.out
+                assert extraction_statement in captured_output.out
+                assert removal_statement in captured_output.out
+                assert totest.os.path.exists(test_path)
+                assert totest.os.path.exists(totest.os.path.join(test_path,\
+                                                                 "test2.txt"))
+                rmtree(test_path) # removed extracted data
```

Comment:
Strictly speaking, the removal isn't needed after the test, but between tests. That's not possible with a fixture.
I only put the removal at the end as well, to easier allow adding more tests, without remembering that the removal is needed between tests.

Reply:
the temporary files are subject to removal at the end of a pytest run independent on whether any test failed. Thus you never need to care about created files to be removed as long as you don't want to recreate them, as we do here.

Reply:
`pytest` temp_dirs are stored for several calls of `pytest` before being deleted. Additionally, leftover files might interfere with other tests, if they end up having the same name. It's cleaner to make sure they're cleaned up after the test.

Reply:
Left over files can't infer with tests in other test functions, because pytest creates a new directory for each test function (that's where a `yield` would act).
The only interference can be for several tests inside the same function, which is the only case, where I do a manual cleanup.

Reply:
I see that it generates new temporary directories for each test function, but it leaves behind the files if the test fails. These are not removed after each pytest invocation. This is going to be a problem when running it locally and this test fails. It might not seem like a problem now, but can be later on. 

You can leave it as is and we can fix it later, when it becomes a problem,

Reply:
This might be a Mac specific issue, on linux even after a failed test pytest will create a new directory and mark the old one as ready to be erased for the OS.